torch_xla/experimental/apply_layers.py

```diff
@@ -0,0 +1,72 @@
+from typing import Iterable
+
+import torch
+import torch.nn as nn
+from torch.utils._pytree import tree_map
+
+from torch_xla.experimental.scan import scan
+
+
+def apply_layers(layers: Iterable[torch.nn.Module], input_data):
+  """Applies each layer in `layers` to `input_data` sequentially.
+
+  `input_data` is provided as input to the first layer in `layers`. The output of one
+  layer is provided as input to next layer. This function is equivalent to
+
+    sequential = torch.nn.Sequential(layers)
+    sequential(input_data)
+
+  This function can be faster to compile since it reuses the XLA computation of the
+  first layer to perform the computation of all other layers.
+  """
+  # Handle empty layers case.
+  try:
+    next(iter(layers))
+  except StopIteration:
+    return input_data
+
+  # Extract and stack the parameters into a pytree.
+  params = [_extract_weights_dict(layer) for layer in layers]
+  stacked_params = tree_map(lambda *tensors: torch.stack(tensors, dim=0),
+                            *params)
+
+  # Use the first layer as the example/template layer.
+  from copy import deepcopy
+  example_layer = deepcopy(next(iter(layers)))
+
+  # Hollow out the weights and biases in the example layer.
+  example_layer = example_layer.to_empty(device=None)
```

Comment:
is this not going to impact the cloned arg?

Reply:
Could you clarify this question -- I thought `to_empty` is going to destroy the value inside `example_layer`, so I `deepcopy` it before to backup.